### PR TITLE
Feature/view iter

### PR DESCRIPTION
**Pull request by evansde77**
@ksnavely @petevg @jcounts @dmannarino 

Implement paged iteration for index objects, clean up some doc strings and pep8.

### CODE REVIEW

File: src/cloudant/database.py [hunk 1]
```diff
@@ -170,6 +184,25 @@ def all_docs(self, **kwargs):
         data = resp.json()
         return data
 
+    @contextlib.contextmanager
+    def custom_index(self, **options):
+        """
+        _custom_index_
```
Comment (ksnavely) [inline src/cloudant/database.py]: cool!

File: src/cloudant/index.py [hunk 2]
```diff
@@ -90,8 +95,48 @@ class Index(object):
...
+
+    Iteration:
+
+    # iterate over all records
+    index = Index()
+    for i in index:
+        print i
+
+    # iterate over records between startkey/endkey
+    index = Index(startkey="2013", endkey="2014")
```
Comment (ksnavely) [inline src/cloudant/index.py]: Missing initial arg? https://github.com/evansde77/metson/pull/12/files#diff-0f234779ef2fcc77f8efea4570248fd4R202

**Comment by ksnavely** [comment]
I think you have a small docs typo but this looks great. LGTM

File: src/cloudant/index.py [hunk 3]
```diff
@@ -90,8 +95,48 @@ class Index(object):
...
+
+    Access by key:
+    index['key'] # get all records matching key
+
+    Slicing by startkey/endkey
+
+    index[["2013","10"]:["2013","11"]] # results between compound keys
+    index["2013":"2014"] # results between string keys
+    index["2013":] # all results after key
+    index[:"2014"] # all results up to key
```
Comment (petevg) [inline src/cloudant/index.py]: My brain is too muzzy today to be of much in terms of critique. But being able to do slices on a couch view really is awesome -- this doc string is exciting to read :-) +1, would read again.

File: src/cloudant/index.py (hunk 2 above)
Comment (evansde77) [inline src/cloudant/index.py]: done

File: src/cloudant/index.py (hunk 3 above)
Comment (evansde77) [inline src/cloudant/index.py]: (thumbs up)

**Pull request merged**